src/adapter/src/coord/sequencer.rs

```diff
@@ -2777,71 +2779,125 @@ impl Coordinator {
             stage,
             format,
             config,
+            no_errors,
             explainee,
         } = plan;
 
+        /// Like [`mz_ore::panic::catch_unwind`], with an extra guard that must be true
+        /// in order to wrap the function call in a [`mz_ore::panic::catch_unwind`] call.
+        fn catch_unwind<R, E, F>(guard: bool, stage: &'static str, f: F) -> Result<R, AdapterError>
+        where
+            F: FnOnce() -> Result<R, E>,
+            E: Into<AdapterError>,
+        {
+            if guard {
+                match mz_ore::panic::catch_unwind(AssertUnwindSafe(f)) {
```

Comment:
CLion is getting confused here about the types for some reason, and makes the entire function red, plus it can't figure out the return type when you call the function. (See screenshot below.)

Could you please factor out the `mz_ore::panic::catch_unwind(AssertUnwindSafe(f))` to a local variable, and explicitly the type (`Result<Result<R, E>, _>`) on the variable definition? I tried this, and it resolves CLion's problem.

![image](https://user-images.githubusercontent.com/11377925/220172128-32fd0b2f-2e4f-4850-af34-511b208aab7d.png)

Reply:
Done.